### DISCUSSION TITLE
Having issue for  .find() when WHERE used.

### DISCUSSION THREAD

**Discussion by ashleshacc** [serverpod/serverpod#578] [Q&A]
@vlidholt @yahu1031 hi, i am trying to get list of data from db where i am checking for a foreign key id, if is working when i try with raw query, but not if i try from .find method if you can please help where im going wrong in the implementation, as far as i have searched, i am doing same as in documentation.

the error i am getting when using .find(), 

flutter: Failed call: journal.getJournalEventList
flutter: ServerpodClientException: Internal server error. Call log id: 42, statusCode = 500
[log] ServerpodClientException: Internal server error. Call log id: 42, statusCode = 500

  Future<List<JournalEvent>> getJournalEventList(
      Session session, int journalId) async {

    var journalList = await JournalEvent.find(
      session,
      where: (t) => t.journalId.equals(journalId),
    );

    return journalList;
  }

and this is the raw .query() which is working for me.

  Future<List<JournalEvent>> getJournalEventList(
      Session session, int journalId) async {

    var journalList = await session.db
        .query("SELECT * FROM journal_event WHERE \"journalId\"=$journalId");

    List<JournalEvent> j = List.empty(growable: true);
    for (var event in journalList) {
      j.add(JournalEvent(
          id: event[0],
          name: event[1],
          journalId: event[2],
          eventType: event[3],
          value: ""));
    }

    return j;
  }

the protocol file

class: JournalEvent
table: journal_event
fields:
  name: String
  journalId: int, parent=journal
  eventType: String
  value: String, api

**Comment by yahu1031** [comment]
Use try-catch to print the error or check the session log to see what the error is. Then we can work on it accordingly @ashleshacc
